### ISSUE TITLE
Not working as intended - it seems there is no config pulling

### ISSUE THREAD

**Issue by nodupe** [LukeSmithxyz/LARBS#42]
Hi,

just tested larbs.sh on a new install of Arch.

Script runs fine, creates the user, install all the programs. When I log in on the newly created user, there are no configuration files on the home directory. In the /root directory, on the other hand, I find a .config folder with ranger configs.

**Comment by dankehidayat** [comment]
I'm not to sure what you mean. I'd just thinking that you don't have all of the config in your /home folder?
Basically this is what i do after fresh installing Arch.

Make sure you install `sudo pacman -S xdg-user-dirs` and run xdg-user-dirs to generate the basic folder [Music,Pictures,Video,etc.]. 
oh yeah, did you type `cd ~` and then `ls -a` ?

**Comment by nodupe** [comment]
No, sorry if I was not clear. What I mean is that the voidrice confs/scripts are not pulled from github and placed inside the .config folder.

**Comment by LukeSmithxyz** [comment]
I'm having a lot of trouble imagining what could've gone wrong on your system. Did you give your user some irregular name?

So you only have ranger configs show up in your /root? No other folders? Is this not a fresh install?

**Comment by nodupe** [comment]
Ok. Fresh install with base and base-dev + xorg + lightdm .

Username larbs.

I’ve run bash larbs.sh, input username and pass, it downloaded all The programs and gives me no error.  

After reboot,  I login from tty and run startx: i3 asks to create the i3 config. 

Super+Enter gives me a term (not st) and ls -la shows me only this newly created i3 config. 

It seems to me that it did not downloaded the configs from github void rice repo. 

I will run larbs.sh again with -r pointing to voidrice.git, with a new user and see if it makes a difference.  I will let you know.

**Comment by LukeSmithxyz** [comment]
Adding the `-r` option will nearly certainly not change things...

...Although it might just be that `git` failed to run the first time around so it will word on re-run. `git` sometimes times out if you have a bad internet connection (or if Github is oer-burdened) and `larbs.sh` doesn't have error handling for this yet. I might have it download the repo earlier and test for it.

And as a general rule, so everyone knows, it's safe to run `larbs.sh` over and over again if something like that goes wrong.

**Comment by nodupe** [comment]
Hi,
I've tried with -r, and as you said, no luck. Internet connection is not an issue. Also did as you suggested and run the script again with same user, no luck.

I've manually cloned the git repos, and after placing all in the right spots, I got a Larbs install.

I am still looking for the culprit. If I can spot it I will let you know.

**Comment by LukeSmithxyz** [comment]
Really strange... I'm curious if other users have had this problem.

Had you created the user earlier before running `LARBS` by any chance?

**Comment by nodupe** [comment]
I just run the script on Arcolinux, and it did all his magic. All installed and configs in place and the usual emoji fonts "bug" that breaks the i3 bar. 

It may as well be that my minimal arch install is missing something to make it work. I will investigate further.

**Comment by LukeSmithxyz** [comment]
Oh I see. What "bug" are you talking about?

**Comment by nodupe** [comment]
1 - I've always created a new user, only did the same user for the test earlier today.
2 - When using arcolinux and larbs the emoji's font gets incredibly huge - so your icons on the bar will not show properly. This happens also in the terminal - for instance if I receive an emoji on a mail on neomutt the same pattern happens.

Example:
https://imgur.com/a/DmNJO4d

It seems to me a conflict in Arco - I was not able to figure out where it is.

**Comment by LukeSmithxyz** [comment]
Okay, I've added a change that will ensure that already created users will have a home that they own. I don't know if this was your original problem or not, but it certainly could have been the cause of your problems or someone else's if they created a user without `-m`.

I'll keep this open for a bit just because if anyone has had this problem I'd hope they say so so I can be sure what it is.

**Comment by nodupe** [comment]
Hi Luke, just a few updates after investigation:

LARBS not installing correctly:  On a fresh install of Arch, I was able install LARBS without issues only after installing the collections base and base-devel. larbs.sh creates user, directories, install apps, and pull the configs and scripts. I'm not sure why I had the problem, but starting from scratch was the solution for me.

Font issue: It was solved only when I replaced ttf-emojione to ttf-emojione-color. There are incompabilities between those fonts, and also notofonts. So after larbs installed ttf-emojione, I've removed it and installed  ttf-emojione-color. Now the bar works properly on ArcolinuxD. Since the font is installed by the packages file, it is a really easy fix for my use case.

**Comment by EISAWESOME** [comment]
I just had the exact same problem.
I was trying out LARBS in virtualbox before installing it to a machine.
After running larbs.sh, the user was created, the programs installed, but the dotfiles were nowhere to be found. Pulling voidrice manually did the trick of course.

**Comment by NagyAttila** [comment]
Same here. After a fresh install using archfi script (https://github.com/MatMoul/archfi), I run the larbs.sh as intented and got the same result. All the pacman packages are installed, user was created, but no dot config files. When I log in to the user and write startx, I got xterm missing error. The user has a simple linux config; only .bashrc.

Is there a way to turn on some debugging mode or logging for the script to catch if there was any error?

**Comment by Awlexus** [comment]
The very same thing just happened to me. After a bit of digging around, I pinpointed the issue to the line where the configs are pulled from. More specifically to sudo -u "$name" I redirected the output from dev/null to a log file and saw "root is not in the sudoers file". Checked the file and only saw the 2 lines added by thr script. I don't know if it happened bacause of the script or because it wasn't there in the first place but after copying the template in sudoers.pacnew it worked and my user's home had some configs. 

_Sent from my K10 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_
